libs/apps/python.go

```diff
@@ -0,0 +1,151 @@
+package apps
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+const DEBUG_PORT = "5678"
+
+var defaultLibraries = []string{
+	"Flask==3.0.3",
+	"streamlit==1.38.0",
+	"gradio==4.44.0",
+	"uvicorn[standard]==0.30.6",
+	"databricks-sdk==0.33.0",
+	"databricks-sql-connector==3.7.3",
+	"debugpy",
+}
+
+type PythonApp struct {
+	config *Config
+	spec   *AppSpec
+	venv   bool
+}
+
+func NewPythonApp(config *Config, spec *AppSpec) *PythonApp {
+	return &PythonApp{config: config, spec: spec}
+}
+
+// PrepareEnvironment prepares the environment for running the app. It first checks if the app is in a virtual environment.
+// If not, it creates a virtual environment and installs the required libraries. It then installs the libraries from
+// requirements.txt if it exists.
+func (p *PythonApp) PrepareEnvironment() error {
+	// First check that we are not already in virtual environment when we execute CLI command
+	// by checking if VIRTUAL_ENV is set
+	if os.Getenv("VIRTUAL_ENV") == "" {
+		// Check if .venv exists
+		_, err := os.Stat(filepath.Join(p.config.AppPath, ".venv"))
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return err
+			}
+
+			err := run([]string{"uv", "venv"}, nil)
```

Comment:
We don't need to create persistent virtual environment, we can use "uv run" to create transient one. It'll be simpler to code and also pretty fast since uv will cache all the dependencies.

Reply:
Indeed, this simplifies code a lot, addressed https://github.com/databricks/cli/pull/2555/commits/aaf77164fe66c4146ba7db8a8af081d030bd2a68